src/eval/datetime.rs

```diff
@@ -0,0 +1,110 @@
+                }
+                InvalidFormatDescription::InvalidModifier { value, .. } => {
+                    format!("modifier {} is invalid", value)
+                }
+                InvalidFormatDescription::Expected { what, .. } => {
+                    format!("expected {}", what)
+                }
+                InvalidFormatDescription::MissingComponentName { .. } => {
+                    "a component name is missing".to_string()
+                }
+                InvalidFormatDescription::MissingRequiredModifier { name, .. } => {
+                    format!("missing required modifier {}", name)
+                }
+                InvalidFormatDescription::NotSupported { context, what, .. } => {
+                    format!("{} is not supported in {}", what, context)
+                }
+                _ => "unable to parse datetime format".to_string(),
+            })?;
+
+        let formatted_result = match self {
+            Datetime::Date(date) => date.format(&format),
+            Datetime::Time(time) => time.format(&format),
+            Datetime::Datetime(datetime) => datetime.format(&format),
+        };
+
+        let unwrapped_result = formatted_result.map_err(|e| match e {
+            Format::InvalidComponent(name) => format!("found invalid component {}", name),
+            _ => "unable to format datetime in the requested format".to_string(),
+        })?;
+
+        Ok(unwrapped_result)
+    }
+
+    pub fn date(&self) -> Option<time::Date> {
+        match self {
+            Datetime::Date(date) => Some(*date),
+            Datetime::Time(_) => None,
+            Datetime::Datetime(datetime) => Some(datetime.date()),
+        }
+    }
+
+    pub fn time(&self) -> Option<time::Time> {
+        match self {
+            Datetime::Date(_) => None,
+            Datetime::Time(time) => Some(*time),
+            Datetime::Datetime(datetime) => Some(datetime.time()),
+        }
+    }
+}
+
+impl Debug for Datetime {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        let year = self.date().map_or("".to_string(), |d| format!("year: {}", d.year()));
+        let month = self
+            .date()
+            .map_or("".to_string(), |d| format!("month: {}", d.month() as u8));
+        let day = self.date().map_or("".to_string(), |d| format!("day: {}", d.day()));
+        let hour = self.time().map_or("".to_string(), |d| format!("hour: {}", d.hour()));
+        let minute = self
+            .time()
+            .map_or("".to_string(), |d| format!("minute: {}", d.minute()));
+        let second = self
+            .time()
+            .map_or("".to_string(), |d| format!("second: {}", d.second()));
+        write!(
+            f,
+            "datetime({})",
+            eco_vec![year, month, day, hour, minute, second]
+                .into_iter()
+                .filter(|e| !e.is_empty())
+                .collect::<EcoVec<String>>()
+                .join(", ")
```

Comment:
you can also use `pretty_array_like` to automatically have fall back to vertical formatting should it get long